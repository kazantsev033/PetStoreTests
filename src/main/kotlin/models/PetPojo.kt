package models

import com.fasterxml.jackson.annotation.JsonIgnoreProperties
import java.util.*

@JsonIgnoreProperties(value = ["id"], allowSetters = true)
data class PetPojo (
	val id: Int = 0,
	val category: Category? = null,
	val name: String = "",
	val photoUrls: List<String> = Collections.emptyList(),
	val tags: List<TagsItem> = Collections.emptyList(),
	val status: String = ""
)

data class TagsItem(val id: Int, val name: String)
data class Category(val id: Int, val name: String)