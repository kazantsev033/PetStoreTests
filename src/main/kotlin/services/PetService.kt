package services

import io.restassured.RestAssured.*
import io.restassured.response.Response
import javax.json.JsonObject

class PetService(baseUrl: String) : RestService(baseUrl) {
    override val basePath: String = "pet"

    fun getPetById(id:String):Response{
        return given()
                .spec(requestSpecification)
                .pathParam("petId",id)
            .`when`()
                .get("$basePath/{petId}")
    }

    fun postPet(petPojo: JsonObject):Response{
        return  given()
                .spec(requestSpecification)
                .body(petPojo.toString())
            .`when`()
                .post(basePath)
    }
}