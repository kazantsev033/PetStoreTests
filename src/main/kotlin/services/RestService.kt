package services

import com.github.viclovsky.swagger.coverage.SwaggerCoverageRestAssured
import io.restassured.builder.RequestSpecBuilder
import io.restassured.specification.RequestSpecification

abstract class RestService (baseUrl:String) {
    abstract val basePath:String
    val requestSpecification:RequestSpecification = RequestSpecBuilder()
        .setBaseUri(baseUrl)
        .addHeader("accept","application/json")
        .addHeader("Content-Type", "application/json")
        .addFilter(SwaggerCoverageRestAssured())
        .build()
}