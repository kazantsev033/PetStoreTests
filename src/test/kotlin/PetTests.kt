import au.com.origin.snapshots.Expect
import au.com.origin.snapshots.junit5.SnapshotExtension
import org.hamcrest.Matchers.containsString
import org.junit.jupiter.api.Test
import org.junit.jupiter.api.extension.ExtendWith
import net.joshka.junit.json.params.JsonFileSource
import org.junit.jupiter.api.TestInfo
import org.junit.jupiter.params.ParameterizedTest
import javax.json.JsonObject
import models.PetPojo
import services.PetService

@ExtendWith(SnapshotExtension::class)
class PetTests: AbstractTests() {
    private val petService = PetService(baseUrl)
    private lateinit var expect: Expect

    @Test
    fun getPetByValidId() {
        val response = petService.getPetById("1")
        response.then().statusCode(200)
        val petPojo = response.`as`(PetPojo::class.java)

        expect.serializer("json").toMatchSnapshot(petPojo)
    }

    @Test
    fun getPetByNonexistentId() {
        val response = petService.getPetById("111")
        response.then().statusCode(404)
        response.then().body(containsString("Pet not found"))
    }

    @Test
    fun getPetByInvalidId() {
        val response = petService.getPetById("test")
        response.then().statusCode(400)
    }

    @ParameterizedTest
    @JsonFileSource(resources = ["testdata/pet/validPets.json"])
    fun postValidPet(json:JsonObject,testInfo: TestInfo){
        petService.postPet(json).then().statusCode(200)
        val responsePet = petService.getPetById(json["id"].toString()).`as`(PetPojo::class.java)


        expect.serializer("json").scenario("${responsePet.name}").toMatchSnapshot(responsePet)
    }
}