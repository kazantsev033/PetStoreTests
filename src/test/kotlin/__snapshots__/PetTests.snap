PetTests.getPetByValidId=[
  {
    "name": "Cat 1",
    "category": {
      "id": 2,
      "name": "Cats"
    },
    "photoUrls": [
      "url1",
      "url2"
    ],
    "tags": [
      {
        "id": 1,
        "name": "tag1"
      },
      {
        "id": 2,
        "name": "tag2"
      }
    ],
    "status": "available"
  }
]


PetTests.postValidPet[John]=[
  {
    "name": "John",
    "category": {
      "id": 1,
      "name": "Fish"
    },
    "photoUrls": [
      "url1,url2"
    ],
    "tags": [
      {
        "id": 1,
        "name": "TestTag1"
      },
      {
        "id": 2,
        "name": "TestTag2"
      }
    ],
    "status": "available"
  }
]


PetTests.postValidPet[Mike]=[
  {
    "name": "Mike",
    "category": {
      "id": 3,
      "name": "Dog"
    },
    "photoUrls": [ ],
    "tags": [ ],
    "status": "sold"
  }
]